src/coreclr/tools/dotnet-pgo/Program.cs

```diff
@@ -1659,13 +1722,24 @@ void AddToInstrumentationData(int eventClrInstanceId, long methodID, int methodF
                     GenerateJittraceFile(commandLineOptions.OutputFileName, methodsUsedInProcess, commandLineOptions.JitTraceOptions);
                 else if (commandLineOptions.FileType.Value == PgoFileType.mibc)
                 {
+                    var config = new MibcConfig();
+
+                    // Look for OS and Arch, e.g. "Windows" and "x64"
+                    TraceEvent processInfo = p.EventsInProcess.Filter(t => t.EventName == "ProcessInfo").FirstOrDefault();
+                    config.Os = processInfo?.PayloadByName("OSInformation")?.ToString();
+                    config.Arch = processInfo?.PayloadByName("ArchInformation")?.ToString();
+
+                    // Look for Sku, e.g. "CoreClr"
+                    TraceEvent runtimeStart = p.EventsInProcess.Filter(t => t.EventName == "Runtime/Start").FirstOrDefault();
```

Comment:
If that eventname is the same as `FireEtwRuntimeInformationDCStart`, it seems to be the right event. I think we can try adding mono to the Sku Enum, right now we are emitting CoreCLR as well on Mono https://github.com/dotnet/runtime/blob/41d9c1cc0dca16743a54971930e81d60777af746/src/mono/mono/eventpipe/ep-rt-mono.c#L2804 since it wasn't clear whether a non 0x1, 0x2 Sku value will cause problems anywhere that its used.

Reply:
I don't see `FireEtwRuntimeInformationDCStart` in my sample nettrace

Reply:
FireEtwRuntimeInformationDCStart is fired as part of rundown, meaning that it will go into every session that gets closed and have requested rundown events (most tools always request rundown events). Currently Mono uses the same Sku enum value as CoreCLR, and was kept that way to not introduce any potential issues in tooling and so far, no need to differentiate the value. If we would like to use the Sku and add a Mono enum value to it, we would need to validate that tools consuming nettrace files are still capable to handle the generated Mono nettrace file (like perfview, TraceEvent library etc).

Reply:
@lateralusX so, just tell me which exact event to check to detect mono runtime? 🙂

Reply:
@lateralusX @EgorBo I'm thinking the value should be 0x4 as 0x3 will show up as DesktopClr, CoreClr since it matches those bits (0x1 || 0x2). I obtained a trace with ` --providers Microsoft-Windows-DotNETRuntime:0x1F000080018:5` on mono and it indeed shows the sku value within the mibc file. https://gist.github.com/mdh1418/2e9ad2d937dbdd0fb0be063c15eff61a

I also verified that we should be able to obtain the actual Sku value from the Mono AOT Compiler side as well. I'll put up a separate PR to integrate the .mibc file validation.

I think we should extend the relevant Sku structs and Mono additions I mentioned earlier

Reply:
Is anything treating the sku as a bit set? Seems like it shouldn't be.

Reply:
@jakobbotsch I'm not sure what the main rationale is between making something a bitmap vs a valuemap, but this might be why its behaving as such https://github.com/dotnet/runtime/blob/4a63cb28b69e1c48bccf592150be7ba297b67950/src/coreclr/vm/ClrEtwAll.man#L611

Reply:
I see, seems reasonable then.